### DISCUSSION TITLE
How do I set up codespace to work for me

### DISCUSSION THREAD

**Discussion by taholmes160** [community/community#72270] [Codespaces] [answered]
### Select Topic Area

Question

### Body

HI folks;

I am currently taking the CS50x course from Harvard and EDx, they use codespaces for their development environment with a web based version of VSCode.

I am also working on a personal coding project in python, and I would like to be able to set up my github account to use codespaces similar to what they are doing with CS50.  

There is just one of me as the developer, and currently, Im running VSCode on my windows computer, running WSL with MariaDB running inside of it.  the codebase right now is about 500 lines and about 15 or 20 templates.  Developing in python with flask.

if its possible (without getting too complex) id like to either run mariadb on the code space (can that be done???) or access the maria db I have on my linux machine, I can reach the db from my windows computer, but the linux computer cannot reach directly to the web.

Im really not sure where to start -- can someone point me in the right direction???

TIM

**Comment by ncastellanosort** [comment, marked as answer]
Hi taholmes160!

Setting up a development environment in GitHub Codespaces for your personal Python project is a great idea, and it can streamline your development process. Look at this steps:

1. **GitHub Repository**:
   - First, make sure your project is hosted on GitHub. If it's not, create a GitHub repository for your project and push your code to it.

2. **Create a Codespace**:
   - Open your GitHub repository and click on the "Code" button, then select "Open with Codespaces." This will create a new Codespace for your project.

3. **Setup Codespace Configuration**:
   - Codespaces use a `.devcontainer` folder for development environment configuration. You can create a `devcontainer.json` file inside this folder to specify the development environment setup.

   - To install MariaDB in your Codespace, you can add a `postCreateCommand` to the `devcontainer.json` file to install it when the Codespace is created.

   - You can also set up any additional extensions, configurations, and dependencies you need in this file.

4. **Accessing External MariaDB**:
   - You mentioned that you want to access your MariaDB instance on your Linux machine. To do this, you need to ensure that your Linux machine allows external database connections.

   - Modify your MariaDB configuration on your Linux machine to listen on external IP addresses, if it's not already configured that way. Make sure to secure your database by setting up firewalls and configuring user access controls.

   - In your Codespace, you can use Python libraries like `mysql-connector-python` or `sqlalchemy` to connect to your external MariaDB. You'll need to provide the IP address or domain of your Linux machine, the database credentials, and the port number to establish a connection.

   - Be cautious about security when connecting to an external database. Use strong passwords, consider SSH tunneling if your Linux machine is not directly accessible from the web, and avoid exposing the database to the internet.

5. **Development**:
   - Once your Codespace is set up and configured, you can start developing your Python project within the Codespace using the web-based VSCode editor.

6. **Commit and Push Changes**:
   - As you make changes to your code, commit them to your GitHub repository and push them to keep your project up to date.

7. **Collaboration**:
   - If you plan to collaborate with others on your project, you can invite collaborators to your GitHub repository. They can also access the Codespace environment if needed.

Hope that this helps!,

Nicolas.

**Reply by taholmes160** [reply]
Thank you! -- i forgot I had this question in, and just re-discovered it!

I'll mess with it and see what I can make happen!

TIM

**Reply by taholmes160** [reply]
HI Nicolas!

I am having problems finding the proper code to do use to install mariadb into my dev container - -google searches are not turning up instructions  and the AI's are helpless in this case, can you please point me to the proper directions?

Thanks
TIM

**Reply by dave-d-dev** [reply]
To run a mariadb (or MySQL) server in GitHub codespaces I used the following commands:
```

sudo apt update

sudo apt install mariadb-server -y

sudo service mariadb start
```

**Comment by yuvayna** [comment]
To set up GitHub Codespaces: make sure you're using a GitHub repository (yours or one you have access to) and that your account has GitHub Pro, Team, or Enterprise. Click the green “Code” button in the repo, go to the “Codespaces” tab, and click “Create codespace on main” (or your desired branch). For a customized environment, add a .devcontainer folder at the root of your repo with a devcontainer.json file like this: 

````json
{
  "name": "My Project",
  "image": "mcr.microsoft.com/devcontainers/javascript-node",
  "postCreateCommand": "npm install",
  "customizations": {
    "vscode": {
      "extensions": ["esbenp.prettier-vscode", "dbaeumer.vscode-eslint"]
    }
  }
}
````

This configures your dev environment automatically with dependencies and extensions. Once the Codespace starts, you'll have a full development IDE in the browser or in VS Code. If you run a local dev server (npm run dev, for example), Codespaces will automatically forward the port and give you a public preview URL. You can manage all your Codespaces at https://github.com/codespaces.
